mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.tmpl

```diff
@@ -13824,6 +13824,84 @@ resource "google_container_cluster" "primary" {
 `, clusterName, networkName, subnetworkName)
 }
 
+func TestAccContainerCluster_withAdvancedMachineFeaturesPMU(t *testing.T) {
+	t.Parallel()
+
+	suffix := acctest.RandString(t, 10)
+	clusterResourceName := "google_container_cluster.primary"
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", suffix)
+	networkName := fmt.Sprintf("test-network-%s", suffix)
+	subnetworkName := fmt.Sprintf("test-subnetwork-%s", suffix)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withAdvancedMachineFeaturesPMU(clusterName, networkName, subnetworkName, "STANDARD"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(clusterResourceName, "node_config.0.advanced_machine_features.0.performance_monitoring_unit", "STANDARD"),
+				),
+			},
+			{
+				Config: testAccContainerCluster_withAdvancedMachineFeaturesPMU(clusterName, networkName, subnetworkName, "ARCHITECTURAL"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(clusterResourceName, "node_config.0.advanced_machine_features.0.performance_monitoring_unit", "ARCHITECTURAL"),
+				),
+			},
+			{
+				Config: testAccContainerCluster_withAdvancedMachineFeaturesPMU(clusterName, networkName, subnetworkName, ""),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(clusterResourceName, "node_config.0.advanced_machine_features.0.performance_monitoring_unit", ""),
+				),
+			},
+			{
+				ResourceName:            clusterResourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"deletion_protection"},
+			},
+		},
+	})
+}
+
+func testAccContainerCluster_withAdvancedMachineFeaturesPMU(clusterName, networkName, subnetworkName, pmuLevel string) string {
+	nodeConfigBlock := fmt.Sprintf(`
```

Comment:
Could we just bake this into the rest of the config string? It's not being used conditionally as far as I can tell.